misaligned.c

```diff
@@ -1,21 +1,42 @@
 #include <stdio.h>
 #include <assert.h>
+#include "misaligned.h"
 
 int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+
     int i = 0, j = 0;
     for(i = 0; i < 5; i++) {
         for(j = 0; j < 5; j++) {
-            printf("%d | %s | %s\n", i * 5 + j, majorColor[i], minorColor[i]);
+            printf("%d | %s | %s\n", i * 5 + j, majorColorNames[i], minorColorNames[j]);
         }
     }
     return i * j;
 }
+ColorPair GetColorFromPairNumber(int pairNumber) {
+    ColorPair colorPair;
+    int zeroBasedPairNumber = pairNumber - 1;
+    colorPair.majorColorValue = 
+        (enum MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
+    colorPair.minorColorValue =
+        (enum MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
+    return colorPair;
+}
+
+void testNumberToPair(int pairNumber,
+    enum MajorColor expectedMajor,
+    enum MinorColor expectedMinor)
+{
+    ColorPair colorPair = GetColorFromPairNumber(pairNumber);
+    assert(colorPair.minorColorValue == expectedMinor);
+    assert(colorPair.majorColorValue == expectedMajor);
+}
 
 int main() {
-    int result = printColorMap();
-    assert(result == 25);
+    printColorMap();
+    testNumberToPair(1,WHITE,BLUE);
+    testNumberToPair(17,YELLOW,ORANGE);
+    testNumberToPair(25,VIOLET,SLATE);
+    
     printf("All is well (maybe!)\n");

```

Comment:
How do we verify alignment of output strings? considering variable length pair number and color names...